src/tri/_tri.cpp

```diff
@@ -672,12 +668,12 @@ py::tuple TriContourGenerator::contour_line_to_segs_and_kinds(const Contour& con
         CodeArray codes(codes_dims);
         unsigned char* codes_ptr = codes.mutable_data();
 
-        for (ContourLine::const_iterator it = contour_line.begin();
-             it != contour_line.end(); ++it) {
-            *segs_ptr++ = it->x;
-            *segs_ptr++ = it->y;
-            *codes_ptr++ = (it == contour_line.begin() ? MOVETO : LINETO);
+        for (const auto & point : contour_line) {
+            *segs_ptr++ = point.x;
+            *segs_ptr++ = point.y;
+            *codes_ptr++ = LINETO;
         }
+        *codes.mutable_data(0) = MOVETO;
```

Comment:
Strictly speaking we should check if `codes` is not empty before calling this. It should never be empty, but if there is a future change that allows it to be then this line could give an access violation. We could either have an explicit check here, or modify the following code block to include it.

Reply:
Yes, good point; added a check.